accepted/0000-subscription-matching-in-public-clouds.md

```diff
@@ -0,0 +1,194 @@
+- GCE: 152986662232938449
+- Azure: 13f56399-bd52-4150-9748-7190aae1ff21
+
+Since the `virtual-host-gatherer` would only deal with the public APIs, each virtual instance will be identified by its "instance id" on the JSON output from the `virtual-host-gatherer` execution. In order to properly match this "instance id" with the "uuid" of a registered system, this will need some ajustments as describer later on this RFC.
+
+### Azure module:
+
+This is an example about how to deal with Azure Public API to collect the virtual instances, using Azure Python SDK:
+
+```python
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.resource import ResourceManagementClient, SubscriptionClient
+
+
+# Tenant ID for your Azure Subscription
+TENANT_ID = 'MY-TENANT-ID'
+
+# Your Service Principal App ID
+CLIENT = 'MY-CLIENT-ID'
+
+# Your Service Principal Password
+KEY = 'MY-SECRET'
+
+credentials = ServicePrincipalCredentials(client_id = CLIENT, secret = KEY, tenant = TENANT_ID)
+subscription_id = 'MY-SUBSCRIPTION-ID'
+compute_client = ComputeManagementClient(credentials, subscription_id)
+
+vmss = compute_client.virtual_machines.list_all()
+for i in vmss:
+    print(i.name)
+```
+
+The response contains the "instance name" as well as the "instance id".
+
+### AWS module:
+
+This is an example about how to deal with AWS EC2 Public API to collect the virtual instances on a particular Zone, using "boto3":
+
+```python
+import boto3
+ec2 = boto3.client("ec2",
+                   region_name='MY-REGION-NAME',
+                   aws_access_key_id='MY-AWS-ACCESS-KEY-ID',
+                   aws_secret_access_key='MY-AWS-ACCESS-KEY-SECRET')
+
+response = ec2.describe_instances()
+for i in response['Reservations']:
+    for j in i['Instances']:
+        print(j)
+```
+
+The response contains the "instance name" as well as the "instance id".
+
+### Google Compute Engine module:
+
+This is an example about how to deal with Google Compute Engine API to collect the virtual instances on a particular Zone, using "google-api-python-client":
+
+```python
+import googleapiclient.discovery
+
+def list_instances(compute, project, zone):
+    result = compute.instances().list(project=project, zone=zone).execute()
+    return result['items'] if 'items' in result else None
+
+    zone = 'MY-ZONE'
+    project = 'MY-PROJECT-NAME'
+
+    compute = googleapiclient.discovery.build('compute', 'v1')
+
+instances = list_instances(compute, project, zone)
+for i in instances:
+    print(i)
+```
+
+The response contains the "instance name" as well as the "instance id".
+
+### A generic JSON-file module:
+
+Currently, there is already a "File based" plugin for `virtual-host-gatherer` which allows to import instances from a custom provided JSON file. According to the existing documentation, the aim of this plugin is to import VMware instances when there is no access from the SUSE Manager server to the VMware.
```

Comment:
Please link to the relevant docs. Is the JSON format identical to the one proposed at the beginning of this RFC?

Reply:
Yes, the JSON example would currently work (except for the reference to `aws` virtualization instance type) for feeding a File-based Virtual Host Manager on SUSE Manager.